### PR TITLE
Add a task to read the performance test times from the performance database

### PR DESCRIPTION
**Pull request by wolfs**
That allows us to update the performance test times file automatically. Instead of using a CSV file we use a JSON file of the following format:

```json
[ {
  "scenario" : "org.gradle.performance.regression.java.JavaIDEModelPerformanceTest.get IDE model for Eclipse",
  "runtimes" : [ {
    "testProject" : "largeMonolithicJavaProject",
    "linux" : 120000
  }, {
    "testProject" : "largeJavaMultiProject",
    "linux" : 476000
  } ]
}, {
  "scenario" : "org.gradle.performance.regression.java.JavaTestChangePerformanceTest.test for non-abi change",
  "runtimes" : [ {
    "testProject" : "mediumJavaMultiProjectWithTestNG",
    "linux" : 291000
  }, {
    "testProject" : "largeJavaMultiProject",
    "linux" : 594000
  }, {
    "testProject" : "largeMonolithicJavaProject",
    "linux" : 1453000
  } ]
} ]
```

This PR also adds storing the class name of the performance test in the database. We need this to properly generate the timings file and as an additional benefit it allows having performance tests with the same method name in different test classes.

The reading of the JSON file is not yet part of this PR, since we first need to populate the database from performance build runs on `master`.

### CODE REVIEW

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        ")\n" +
+        "select\n" +
+        "last.testClass,\n" +
+        "last.testId,\n" +
+        "last.testProject,\n" +
+        "testExecution.startTime,\n" +
+        "testExecution.endTime\n" +
+        "from last\n" +
+        "join testExecution on testExecution.id = last.lastId\n" +
+        "order by last.testClass,last.testId,last.testProject";
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java:62]: Can this be a groovy file so we can have more readable multi-line string?

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        });
+    }
+
+    protected <RESULT> RESULT withConnection(String actionName, ConnectionAction<RESULT> action) {
+        return db.withConnection(actionName, action);
+    }
+
+
+        @Override
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java:99]: Nit: seems like incorrect indent.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceTestRuntimesGenerator.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    "scenario", scenario.getClassName() + "." + scenario.getTestName(),
+                    "runtimes", times.stream()
+                        .map(experimentEntry -> ImmutableMap.of(
+                            "testProject", experimentEntry.getKey().getTestProject(),
+                            "linux", experimentEntry.getValue()
+                        ))
+                        .collect(Collectors.toList())
+                );
+            })
+            .collect(Collectors.toList());
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceTestRuntimesGenerator.java:59]: Any reason why not declaring extra Java Beans? `Map`s are hard to maintain in the long term...

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/ResultsStore.java [hunk 4]
```diff
@@ -34,4 +35,6 @@
      * Returns the n most recent instances of the given test which are younger than the max age.
      */
     PerformanceTestHistory getTestResults(PerformanceExperiment experiment, int mostRecentN, int maxDaysOld, String channel);
+
+    Map<PerformanceExperiment, Long> getEstimatedExperimentTimes(OperatingSystem operatingSystem);
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/ResultsStore.java:39]: I would name it `getEstimatedExperimentTimesMillis` or `getEstimatedExperimentTimesSecond` to clearly indicate what's inside.

**Review by blindpirate — COMMENTED** [review]
You've fixed so many tiny issues I've been thinking for a long time, like `try-with-resources`, getting time from DB, thanks for the cleanup!

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/ResultsStore.java (hunk 4 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/ResultsStore.java:39]: Fixed in https://github.com/gradle/gradle/pull/14672

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceTestRuntimesGenerator.java (hunk 3 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceTestRuntimesGenerator.java:59]: Fixed in https://github.com/gradle/gradle/pull/14672

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java (hunk 2 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java:99]: Fixed in https://github.com/gradle/gradle/pull/14672

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java (hunk 1 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java:62]: Converted to Groovy in https://github.com/gradle/gradle/pull/14672.

**Pull request merged**